modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinQueryMemoryTrackerSelfTest.java

```diff
@@ -77,7 +78,9 @@ public void testPartialResultRead() throws Throwable {
             }
         }
 
-        assertEquals(localResults.size(), closedResultCntr.get());
+        // A wait is required because the query cancel request handler runs asynchronously.
+        boolean success = waitForCondition(() -> localResults.size() == closedResultCntr.get(), getTestTimeout());
```

Comment:
I would prefer small fixed timeout for such case, we should't wait 5 mins in case any problem, Le'ts use for example 10 secs

Reply:
Done